### PR TITLE
Remove unneeded unconditional jumps for Arm/Arm64

### PR DESCRIPTION
**Pull request by kunalspathak**
Extend https://github.com/dotnet/runtime/pull/69041 for Arm/Arm64.

cc: @Wraith2

### CODE REVIEW

File: src/coreclr/jit/emit.cpp [hunk 1]
```diff
@@ -4189,8 +4185,12 @@ void emitter::emitRemoveJumpToNextInst()
         insGroup*     jmpGroup = jmp->idjIG;
         instrDescJmp* nextJmp  = jmp->idjNext;
 
-        if (jmp->idInsFmt() == IF_LABEL && emitIsUncondJump(jmp) && jmp->idjIsRemovableJmpCandidate)
+        if (emitIsUncondJump(jmp) && jmp->idjIsRemovableJmpCandidate)
```
[the remark this answers is not in the document]
Comment (kunalspathak) [reply src/coreclr/jit/emit.cpp:4188]: Ah ok. Let me put it back.

File: src/coreclr/jit/emit.cpp [hunk 2]
```diff
@@ -1006,14 +1006,19 @@ insGroup* emitter::emitSavIG(bool emitAdd)
...
-
-#if defined(TARGET_XARCH)
         assert(emitLastIns != nullptr);
-        if (emitLastIns->idIns() == INS_jmp)
+
+#ifdef TARGET_XARCH
+        instruction jmpInstr = INS_jmp;
+#else // TARGET_ARM64
+        instruction jmpInstr = INS_b;
+#endif
```
Comment (MichalPetryka) [inline src/coreclr/jit/emit.cpp:1015]: Won't the assumption that it's either ARM or XARCH be invalid on LoongArch64 builds?

Comment (kunalspathak) [reply src/coreclr/jit/emit.cpp:1015]: they will be and I need to do fix/validate it. This is draft and I would like to get a sense of its state before I spend time on other archs.

**Comment by kunalspathak** [comment]
As expected, Arm needs some more fixes, will do it when I get a chance.

**Pull request closed without merging**